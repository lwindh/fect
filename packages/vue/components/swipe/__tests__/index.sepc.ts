import { flushPromises, mount } from '@vue/test-utils'
import Swipe from '..'
import SwipeItem from '../../swipe-item'
import { later } from '../../../tests'

const Wrapper = {
  components: {
    [Swipe.name]: Swipe,
    [SwipeItem.name]: SwipeItem,
  },
  data() {
    return {
      loop: false,
      autoplay: 0,
      indicatorDisplay: true,
      indicatorColor: '',
      initialValue: 0,
    }
  },
  template: `
   <div class="container" >
    <fe-swipe 
    :loop="loop" 
    :autoplay="autoplay" 
    :indicator-display="indicatorDisplay"
    :indicator-color="indicatorColor"
    :initial-value="initialValue"
    style="height:200px;width:350px;"
    >
      <fe-swipeItem>
        1
      </fe-swipeItem>
      <fe-swipeItem>
        2
      </fe-swipeItem>
      <fe-swipeItem>
        3
      </fe-swipeItem>
    </fe-swipe>
   </div>
  `,
}

describe('Swipe', () => {
  it('should be render as a element', async () => {
    const wrapper = mount(Wrapper)
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
    wrapper.unmount()
  })
  it('should be support custom indicator color', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({ indicatorColor: 'red' })
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  it('component props should be work correctly', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({
      loop: true,
      indicatorDisplay: false,
      autoplay: 2000,
    })
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })
  it('component should processing boundary index correctly', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({
      loop: true,
      autoPlay: 2000,
      initialValue: -1,
    })
    await flushPromises()
    const indicators = wrapper.findAll('.fect-swipe__indicator')
    await indicators[0].trigger('click')
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
  })

  it('component swipeItem should emitter a click event', async () => {
    const wrapper = mount(Wrapper)
    await flushPromises()
    const els = wrapper.findAll('.fect-swipe-item')
    console.log(els)
    await els[0].trigger('click')
    expect(wrapper.emitted()).toBeTruthy()
    wrapper.unmount()
  })
  it('should support autoplay', async () => {
    const wrapper = mount(Wrapper)
    await wrapper.setData({
      loop: true,
      autoplay: 2000,
    })
    // await flushPromises()
    // await later()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
