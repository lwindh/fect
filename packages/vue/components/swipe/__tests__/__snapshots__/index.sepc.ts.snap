// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swipe component props should be work correctly 1`] = `
<div class="container">
  <div class="fect-swipe" style="height: 200px; width: 350px;">
    <div class="fect-swipe__track" style="width: 1050px; transform: translateX(0px); transition-duration: 0ms;">
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 1 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 2 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 3 </div>
    </div>
    <!---->
  </div>
</div>
`;

exports[`Swipe component should processing boundary index correctly 1`] = `
<div class="container">
  <div class="fect-swipe" style="height: 200px; width: 350px;">
    <div class="fect-swipe__track" style="width: 1050px; transform: translateX(1400px); transition-duration: 0ms;">
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 1 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 2 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 3 </div>
    </div>
    <div class="fect-swipe__indicators"><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 1;"></span><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 0.3;"></span><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 0.3;"></span></div>
  </div>
</div>
`;

exports[`Swipe should be render as a element 1`] = `
<div class="container">
  <div class="fect-swipe" style="height: 200px; width: 350px;">
    <div class="fect-swipe__track" style="width: 1050px; transform: translateX(0px); transition-duration: 0ms;">
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 1 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 2 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 3 </div>
    </div>
    <div class="fect-swipe__indicators"><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 1;"></span><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 0.3;"></span><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 0.3;"></span></div>
  </div>
</div>
`;

exports[`Swipe should be support custom indicator color 1`] = `
<div class="container">
  <div class="fect-swipe" style="height: 200px; width: 350px;">
    <div class="fect-swipe__track" style="width: 1050px; transform: translateX(0px); transition-duration: 0ms;">
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 1 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 2 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 3 </div>
    </div>
    <div class="fect-swipe__indicators"><span class="fect-swipe__indicator" style="background-color: red; width: 8px; height: 8px; opacity: 1;"></span><span class="fect-swipe__indicator" style="background-color: red; width: 8px; height: 8px; opacity: 0.3;"></span><span class="fect-swipe__indicator" style="background-color: red; width: 8px; height: 8px; opacity: 0.3;"></span></div>
  </div>
</div>
`;

exports[`Swipe should support autoplay 1`] = `
<div class="container">
  <div class="fect-swipe" style="height: 200px; width: 350px;">
    <div class="fect-swipe__track" style="width: 1050px; transform: translateX(0px); transition-duration: 0ms;">
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 1 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 2 </div>
      <div class="fect-swipe-item" style="width: 350px; transform: translateX(0px);"> 3 </div>
    </div>
    <div class="fect-swipe__indicators"><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 1;"></span><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 0.3;"></span><span class="fect-swipe__indicator" style="width: 8px; height: 8px; opacity: 0.3;"></span></div>
  </div>
</div>
`;
